prov/gni/src/gnix_ep.c

```diff
@@ -1371,10 +1412,46 @@ DIRECT_FN STATIC int gnix_ep_control(fid_t fid, int command, void *arg)
 					ret);
 				goto err;
 			}
-			if (ep->send_cq)
+
+			ret = _gnix_ep_enable(ep);
+			if (ret != FI_SUCCESS) {
+				GNIX_WARN(FI_LOG_EP_CTRL,
+				     "_gnix_ep_enable call returned %d\n",
+					ret);
+				goto err;
+			}
+
+			/*
+			 * set enable flags and add nic to poll lists
+			 */
+			if (ep->send_cq) {
+				_gnix_cq_poll_nic_add(ep->send_cq, ep->nic);
 				ep->tx_enabled = true;
-			if (ep->recv_cq)
+			}
+			if (ep->recv_cq) {
+				_gnix_cq_poll_nic_add(ep->recv_cq, ep->nic);
 				ep->rx_enabled = true;
+			}
+			if (ep->send_cntr)
+				_gnix_cntr_poll_nic_add(ep->send_cntr, ep->nic);
+
+			if (ep->recv_cntr)
+				_gnix_cntr_poll_nic_add(ep->recv_cntr, ep->nic);
+
+			if (ep->write_cntr)
+				_gnix_cntr_poll_nic_add(ep->write_cntr,
+							ep->nic);
+
+			if (ep->read_cntr)
+				_gnix_cntr_poll_nic_add(ep->read_cntr, ep->nic);
+
+			if (ep->rwrite_cntr)
+				_gnix_cntr_poll_nic_add(ep->rwrite_cntr,
+							ep->nic);
+
+			if (ep->rread_cntr)
+				_gnix_cntr_poll_nic_add(ep->rread_cntr,
+							ep->nic);
```

Comment:
Did you mean to remove this section after adding the call to the new function _gnix_ep_enable?